libvirt/tests/src/virtual_network/qemu/nic_bonding.py

```
@@ -0,0 +1,123 @@
+    session_serial.cmd_output_safe("ip link set dev bond0 addr %s up" % mac)
+    setup_cmd = "ifenslave bond0 " + " ".join(ifnames)
+    session_serial.cmd_output_safe(setup_cmd)
+    dhcp_cmd = params.get("dhcp_cmd")
+    session_serial.cmd_output_safe(dhcp_cmd, timeout=240)
+    # prepare test data
+    guest_path = os.path.join(tmp_dir, "dst-%s" % utils_misc.generate_random_string(8))
+    host_path = os.path.join(
+        test.tmpdir, "tmp-%s" % utils_misc.generate_random_string(8)
+    )
+    test.log.info("Test setup: Creating %dMB file on host", filesize)
+    process.run(dd_cmd % host_path, shell=True)
```

Ruff (0.14.8)

Function call with `shell=True` parameter identified, security issue

(S604)